packages/react-components/src/components/CheckboxFacet/index.tsx

```diff
@@ -0,0 +1,135 @@
+  isMobile?: boolean;
+  /** Callback invoked on search input change */
+  onSearch: (evt: ChangeEvent<HTMLInputElement>) => any;
+  /** Callback invoked on request to expand list completely */
+  onToggle: (evt: Event) => any;
+
+  hidden: boolean;
+}
+
+export default ({
+  theme = styles,
+  config,
+  isMobile,
+  facet,
+  hidden,
+}: ICheckboxFacetProps) => {
+  const [search, setSearch] = useState('');
+  const [isExpanded, setExpanded] = useState(false);
+  const translate = useTranslations();
+
+  const onSearch = useCallback((e: React.FormEvent<HTMLInputElement>) => {
+    setSearch(e.target ? e.target.value : e);
+  }, []);
+
+  const items = useMemo(() => {
+    if (isExpanded && search) {
+      const regexp = new RegExp(escapeRegExp(search), 'gi');
+      return facet.get('values').filter((i) => regexp.test(i.get('value')));
+    }
+    return facet.get('values');
+  }, [search, isExpanded, facet]);
+
+  return (
+    <div
+      className={cx(theme.root, { [theme.mobile]: isMobile })}
+      id={`facet-${facet.get('name')}`}
+      role="region"
+      hidden={hidden}
+    >
+      <div className={theme.search} display-if={isExpanded}>
+        <input
+          placeholder={t('facets.search')}
+          className={theme.input}
+          onChange={onSearch}
+          value={search}
+        />
+        <Icon name="Search" className={theme.icon} title={t('facets.search')} />
+      </div>
+
+      <section role="list">
+        <MapArray
+          display-if={config.get('pullSelected')}
+          array={items.filter((i) => i.get('selected'))}
+          factory={Checkbox}
+          content={content}
+          isMobile={isMobile}
+        />
+
+        <VirtualizedList
+          display-if={isExpanded}
+          factory={Checkbox}
+          isMobile={isMobile}
+          onItemClick={() => setSearch('')}
+          config={config}
+          content={content}
+          limit={config.get('maxItemsCount', 6)}
+          className={theme.expandedList}
+          array={
+            config.get('pullSelected')
+              ? items.filter((i) => !i.get('selected'))
+              : items
+          }
+        />
+
+        <MapArray
+          display-if={!isExpanded}
+          factory={Checkbox}
+          isMobile={isMobile}
+          limit={!isMobile && config.get('maxItemsCount', 6)}
+          content={content}
+          keyAccessor={(i) => i.get('value')}
+          array={
+            config.get('pullSelected')
+              ? items.filter((i) => !i.get('selected'))
+              : items
+          }
+        />
+      </section>
+
+      <Button
+        className={theme.expand}
+        onClick={() => setExpanded((exp) => !exp)}
+        display-if={!isMobile && items.size > config.get('maxItemsCount', 6)}
+      >
+        <Text primary uppercase>
+          <Icon
+            name={isExpanded ? 'Minus' : 'Plus'}
+            title={translate(isExpanded ? 'facets.expanded' : 'Collapsed')}
```

Comment:
@Bitaru here should be `facets.collapsed` i think ?

Reply:
😩